### PR TITLE
e2e(quasar-cli-vue3-webpack-javascript sample)

### PR DESCRIPTION
**Pull request by Andranik-mikhaielian**
Screenshot 2023-01-04 at 2 49 29 PM

### CODE REVIEW

File: cypress/common/selectors.ts [hunk 1]
```diff
@@ -38,7 +39,12 @@ export const baseSelectors = {
     body: 'body',
     unorderedList: 'ul',
     span: 'span',
-    checkbox: 'input[type="checkbox"]'
+    checkbox: 'input[type="checkbox"]',
+    appExposesCounter: '[data-cy="app-button-counter"]',
+    appExposesButton: '[data-cy="app-button"]',
+    appExposesNames: '[data-e2e="exposesAppNames"]',
+    appExposesCloseName: '[data-e2e="exposesAppNamesClose"]',
+    appGeneralCounter: '[data-e2e="General-counter"]'
```
Comment (PeaceDate) [inline cypress/common/selectors.ts]: Please, transfer these locators to corresponding blocks

File: cypress/fixtures/constants.ts [hunk 2]
```diff
@@ -408,6 +408,24 @@ export class Constants {
...
+        appGeneralBanner: 'App Remote Home Page',
+        appGeneralRouteButton: 'Route',
+        appGeneralSubheader1: 'Exposed from Child',
+        appGeneralSubheader2: 'Listening in Parent',
+        appGeneralCounter: 'Counter',
+        appGeneralName1: 'Riyaz',
+        appGeneralName2: 'Quy',
+        appGeneralName3: 'Sang',
+        appGeneralName4: 'Loris',
+
```
Comment (PeaceDate) [inline cypress/fixtures/constants.ts]: Update to this approach:
```
quasarCli: {
   appExposes: {
   // only app exposes elements here
      list: 'List',
      banner: 'App Exposes Home Page',
      ....
   },
   appGeneral: {
   // only app general elements here
      name: 'Child FE',
      banner: 'App Remote Home Page',
      ...
   },
   // common elements here
   button: 'App Button',
}
```

Then use it like that: 

```
Constants.elementText.quasarCli.button - for common elements
Constants.elementText.quasarCli.AppExposes.list - for app exposes elements
Constants.elementText.quasarCli.AppGeneral.list - for app general elements
```

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+            })
+            // basePage.openLocalhost(3002)
+            // basePage.clickElementWithText({
+            //     selector: baseSelectors.linkTag,
+            //     text: String(property.appExposesComponentsButton)
+            // })
+            // basePage.checkElementQuantity({
+            //     selector: baseSelectors.appExposesNames,
+            //     quantity: 5
+            // })
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: Remove comments

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        })
+    })
+})
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: Add BL

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+            basePage.checkElementWithTextPresence({
+                selector: baseSelectors.h4,
+                text: String(property.appExposesBanner)
+            })
+            basePage.clickElementBySelector({
+                selector: baseSelectors.button
+            })
+            basePage.clickElementBySelector({
+                selector: baseSelectors.button
+            })
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts:86]: Why do you click on button for the second time?

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+            basePage.clickElementWithText({
+                selector: baseSelectors.button,
+                text: String(property.clickMeButton)
+            })
+            basePage.checkElementWithTextPresence({
+                selector: baseSelectors.appExposesCounter,
+                text: '1'
+            })
+            basePage.openLocalhost(3001)
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: For reload you can use `reloadWindow` method

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+            })
+            basePage.clickElementBySelector({
+                selector: baseSelectors.appExposesCloseName,
+                index: 1
+            })
+            basePage.checkElementQuantity({
+                selector: baseSelectors.appExposesNames,
+                quantity: 1
+            })
+            basePage.openLocalhost(3001)
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: Same

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+            basePage.checkElementWithTextPresence({
+                selector: baseSelectors.h4,
+                text: String(property.appGeneralBanner)
+            })
+            basePage.clickElementBySelector({
+                selector: baseSelectors.button
+            })
+            basePage.clickElementBySelector({
+                selector: baseSelectors.button
+            })
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts:157]: Same) Why second click?

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+                selector: baseSelectors.linkTag,
+                text: String(property.appExposesComponentsButton),
+                isForce: true
+            })
+            basePage.clickElementWithText({
+                selector: baseSelectors.linkTag,
+                text: String(property.appGeneralRouteButton),
+                isForce: true
+            })
+            basePage.checkUrlText('http://localhost:3002/#/route', true)
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: Add url string to Constants

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+            basePage.clickElementWithText({
+                selector: baseSelectors.linkTag,
+                text: String(property.appExposesComponentsButton),
+                isForce: true
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: You really need force here?

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+                text: String(property.appListDiv)
+            })
+            basePage.checkElementQuantity({
+                selector: baseSelectors.appExposesNames,
+                quantity: 5
+            })
+            basePage.checkElementQuantity({
+                selector: baseSelectors.appExposesNames,
+                quantity: 5
+            })
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts]: Do you need second check?

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts [hunk 12]
```diff
@@ -0,0 +1,302 @@
+import {BaseMethods} from "../../cypress/common/base";
```
Comment (PeaceDate) [inline quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts:1]: Try to think more how you can simplify this test) I think there is route to make it more smaller)

**Comment by PeaceDate** [comment]
And rebase branch please

**Comment by PeaceDate** [comment]
Change name to e2e(quasar-cli-vue3-webpack-javascript sample)

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts (hunk 5 above)
Comment (Andranik-mikhaielian) [reply quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts:86]: to open the left side menu and reach the Components button, cause when I click for the first time -> the Left side menu is hiding, also for check this Hamburger button

Comment (Andranik-mikhaielian) [reply quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts:86]: Screenshot 2023-01-05 at 3 10 58 PM
Screenshot 2023-01-05 at 3 11 05 PM

File: quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts (hunk 8 above)
Comment (Andranik-mikhaielian) [reply quasar-cli-vue3-webpack-javascript/e2e/CheckAppExposes.cy.ts:157]: same reason) "to open the left side menu and reach the Components button, cause when I click for the first time -> the Left side menu is hiding, also for check this Hamburger button"

File: cypress/common/selectors.ts [hunk 13]
```diff
@@ -38,7 +39,7 @@ export const baseSelectors = {
     body: 'body',
     unorderedList: 'ul',
     span: 'span',
-    checkbox: 'input[type="checkbox"]'
+    checkbox: 'input[type="checkbox"]',
```
Comment (PeaceDate) [inline cypress/common/selectors.ts]: remove comma in the end

File: cypress/common/selectors.ts [hunk 14]
```diff
@@ -69,6 +70,11 @@ export const selectors = {
         starSymbol: '[data-e2e="STAR__SYMBOL"]',
         cloudSymbol: '[data-e2e="CLOUD__SYMBOL"]',
     },
+    appExposesCounter: '[data-cy="app-button-counter"]',
+    appExposesNames: '[data-e2e="exposesAppNames"]',
+    appExposesCloseName: '[data-e2e="exposesAppNamesClose"]',
+    appGeneralCounter: '[data-e2e="General-counter"]',
+    appExposesButton: '[data-cy="app-button"]'
```
Comment (PeaceDate) [inline cypress/common/selectors.ts]: move it to the buttons section

File: cypress/fixtures/constants.ts [hunk 15]
```diff
@@ -415,6 +415,29 @@ export class Constants {
...
+                counter: 'Counter',
+                name1: 'Riyaz',
+                name2: 'Quy',
+                name3: 'Sang',
+                name4: 'Loris'
+            }
+        },
+        appButtonDiv: 'App Button',
+        appButtonClickMeButton: 'Click me',
+        appListDiv: 'AppList',
```
Comment (PeaceDate) [inline cypress/fixtures/constants.ts]: Move these three locators to the `quasarCli ` object

```
        quasarCli: {
            appExposes: {
                name: 'App Exposes',
                list: 'List',
                banner: 'App Exposes Home Page',
                componentsButton: 'Components'
            },
            appGeneral: {
                name: 'Child FE',
                banner: 'App Remote Home Page',
                routeButton: 'Route',
                subheader1: 'Exposed from Child',
                subheader2: 'Listening in Parent',
                counter: 'Counter',
                name1: 'Riyaz',
                name2: 'Quy',
                name3: 'Sang',
                name4: 'Loris'
            },
            appButtonDiv: 'App Button',
            appButtonClickMeButton: 'Click me',
            appListDiv: 'AppList',
        },
```

File: cypress/common/selectors.ts (hunk 14 above)
Comment (PeaceDate) [reply cypress/common/selectors.ts]: `appExposesButton` only

**Pull request merged**